packages/expo-cli/src/commands/eas-build/build/action.ts

```diff
@@ -145,7 +111,13 @@ async function startBuildAsync(
     await builder.ensureCredentialsAsync();
 
     if (!builder.ctx.skipProjectConfiguration) {
-      await builder.configureProjectAsync();
+      if (builder instanceof iOSBuilder) {
+        await builder.configureProjectAsync();
+      } else if (!(await builder.isProjectConfiguredAsync())) {
+        throw new CommandError(
+          'Project is not configured. Please run "expo eas:build:init" first to configure the project'
+        );
+      }
```

Comment:
```suggestion
      await builder.enureProjectConfiguredAsync();
```
and `enureProjectConfiguredAsync` on android would  contain only
```
     if (!(await this.isProjectConfiguredAsync())) {
        throw new CommandError(
          'Project is not configured. Please run "expo eas:build:init" first to configure the project'
        );
      }
```

I would suggest replacing `configureProjectAsync` -> `enureProjectConfiguredAsync` in the builder interface

configureProjectAsync would be a public method on builders but not the part of the interface

when building you would call `enureProjectConfiguredAsync` methods
when running init you can call directly `configureProjectAsync`

Reply:
as a side note
we will probably want to remove project configuration and credentials setup from the builder in the future, this interface should be kept as small as posible